packages/vscode-extension/src/webview/providers/UtilsProvider.tsx

```diff
@@ -34,6 +36,36 @@ export function useUtils() {
   return context;
 }
 
+type TelemetryContextProps = { telemetryEnabled: boolean };
+
+const TelemetryContext = createContext<TelemetryContextProps>({ telemetryEnabled: false });
+
+export function TelemetryProvider({ children }: PropsWithChildren) {
```

Comment:
IMO, it should be in a separate file. I am not sure why we need to pass utils through the context, but if we cannot just import it in a separate file, you can always use `useUtils` in `TelemetryProvider`. Just remember to order providers properly (imo this is why it would be better to have it just in hook instead of context, especially since we don't use it anywhere else).